tf/examples/Bonsai/helpermethods.py

```diff
@@ -143,48 +156,60 @@ def preProcessData(dataDir):
 
     Xtrain = train[:, 1:dataDimension + 1]
     Ytrain_ = train[:, 0]
-    numClasses = max(Ytrain_) - min(Ytrain_) + 1
+    Ytrain = Ytrain_
 
     Xtest = test[:, 1:dataDimension + 1]
     Ytest_ = test[:, 0]
-
-    numClasses = int(max(numClasses, max(Ytest_) - min(Ytest_) + 1))
+    Ytest = Ytest_
 
     # Mean Var Normalisation
     mean = np.mean(Xtrain, 0)
     std = np.std(Xtrain, 0)
     std[std[:] < 0.000001] = 1
     Xtrain = (Xtrain - mean) / std
-
     Xtest = (Xtest - mean) / std
     # End Mean Var normalisation
 
-    lab = Ytrain_.astype('uint8')
-    lab = np.array(lab) - min(lab)
+    # Classification.
+    if (isRegression == False):
+        numClasses = max(Ytrain_) - min(Ytrain_) + 1
+        numClasses = int(max(numClasses, max(Ytest_) - min(Ytest_) + 1))
 
-    lab_ = np.zeros((Xtrain.shape[0], numClasses))
-    lab_[np.arange(Xtrain.shape[0]), lab] = 1
-    if (numClasses == 2):
-        Ytrain = np.reshape(lab, [-1, 1])
-    else:
-        Ytrain = lab_
+        lab = Ytrain_.astype('uint8')
+        lab = np.array(lab) - min(lab)
 
-    lab = Ytest_.astype('uint8')
-    lab = np.array(lab) - min(lab)
+        lab_ = np.zeros((Xtrain.shape[0], numClasses))
+        lab_[np.arange(Xtrain.shape[0]), lab] = 1
+        if (numClasses == 2):
+            Ytrain = np.reshape(lab, [-1, 1])
+        else:
+            Ytrain = lab_
 
-    lab_ = np.zeros((Xtest.shape[0], numClasses))
-    lab_[np.arange(Xtest.shape[0]), lab] = 1
-    if (numClasses == 2):
-        Ytest = np.reshape(lab, [-1, 1])
-    else:
-        Ytest = lab_
+        lab = Ytest_.astype('uint8')
+        lab = np.array(lab) - min(lab)
+
+        lab_ = np.zeros((Xtest.shape[0], numClasses))
+        lab_[np.arange(Xtest.shape[0]), lab] = 1
+        if (numClasses == 2):
+            Ytest = np.reshape(lab, [-1, 1])
+        else:
+            Ytest = lab_
+
+    elif (isRegression == True):
```

Comment:
Throw an warning here if numclasses is not  one.

Reply:
I set the number of classes to 1 here, so no need of a warning here.